js/scripts.js

```diff
@@ -37,8 +14,10 @@ let pokemonRespository = (function() {
         });
     }
 
-    function add(newPokemon) {
-        if (typeof newPokemon === "object" && JSON.stringify(Object.keys(newPokemon)) === JSON.stringify(Object.keys(pokemonList[0]))) {
+    // Adds pokémon entry to pokemonList
+    function addEntry(newPokemon) {
+        // Checks whether the entry is an object and includes the expected keys
+        if (typeof newPokemon === "object" && JSON.stringify(Object.keys(newPokemon)) === JSON.stringify(objectKeys)) {
```

Comment:
This works, but we could create a helper function using `Array.every()` method. You can also checkout `Array.some()` to check for some keys
```
const requiredKeys = (pokemon, keys) => keys.every(key => key in pokemon);
```

```
function addEntry(newPokemon) {
        if (requiredKeys(newPokemon, ['name', 'detailsUrl'])) {
           return pokemonList.push(newPokemon);
        } else {
            // TODO
        }
    }
```

Reply:
Interesting, this works, but there's something I'm not understanding about arrow functions. When I try to convert your `requiredKeys` function into a function definition, it doesn't work (the comparison returns `false`:
`    function requiredKeys(pokemon, keys) {
        return keys.some(function (key) {
            key in pokemon
        });
    }
`
What's going on?

Reply:
Was missing `return`